tests/check-regex-filtering/src/lib.rs

```diff
@@ -0,0 +1,45 @@
+/// This test is compiled by the Github workflows with the
```

Comment:
This looks fantastic, exactly what I hoped for.

Could you either either document the way that this checks the recursive timezone check, or change the filter to filter out something that would only be filtered out if the filter applied recursively? For example, changing the filter to `Europe` and then asserting that at least some European countries do not appear. (Assuming that that's a correct implication of the hierarchy, I'm not particularly familiar with all the DB rules.)

Reply:
I've added a few more tests and some comments explaining what I think should be included and excluded. This stuff does get tricky...if we take your example of 'Europe' for example you get some Asian timezones - because of Istanbul! It all depends what's in the links, and I guess that is subject to change so we should be careful of how restrictive we are.